File: app/src/main/java/com/example/producttest/activity/HomeActivity.java
package com.example.producttest.activity;

import androidx.annotation.NonNull;
import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentTransaction;
import androidx.viewpager.widget.ViewPager;

import android.os.Bundle;
import android.util.Log;
import android.view.MenuItem;

import com.example.producttest.R;
import com.example.producttest.adapter.ViewPagerHomeAdapter;
import com.example.producttest.fragments.CartFragment;
import com.example.producttest.fragments.HistoryFragment;
import com.example.producttest.fragments.ProductListFragment;
import com.google.android.material.bottomnavigation.BottomNavigationView;

public class HomeActivity extends AppCompatActivity implements BottomNavigationView.OnNavigationItemSelectedListener {
    private BottomNavigationView bottomNavigationView;
    private FragmentTransaction fragmentTransaction;
    private ProductListFragment productListFragment;
    private CartFragment cartFragment;
    private HistoryFragment historyFragment;
    MenuItem menuItem;
    private ViewPager viewPager;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_home);
        bottomNavigationView = findViewById(R.id.nav_views);
        bottomNavigationView.setOnNavigationItemSelectedListener(this);

        viewPager = (ViewPager) findViewById(R.id.viewpager);
        viewPager.addOnPageChangeListener(new ViewPager.OnPageChangeListener() {
            @Override
            public void onPageScrolled(int position, float positionOffset, int positionOffsetPixels) {
            }
            @Override
            public void onPageSelected(int position) {
                if (menuItem != null) {
                    menuItem.setChecked(false);
                }
                else
                {
                    bottomNavigationView.getMenu().getItem(0).setChecked(false);
                }
                Log.d("pagess", "onPageSelected: "+position);
                bottomNavigationView.getMenu().getItem(position).setChecked(true);
                menuItem = bottomNavigationView.getMenu().getItem(position);
            }

            @Override
            public void onPageScrollStateChanged(int state) {

            }
        });
        setupViewPager(viewPager);
    }

    public ViewPager getViewPager() {
        if (null == viewPager) {
            viewPager = (ViewPager) findViewById(R.id.viewpager);
        }
        return viewPager;
    }
    private void setupViewPager(ViewPager viewPager) {
        ViewPagerHomeAdapter adapter = new ViewPagerHomeAdapter(getSupportFragmentManager());
        productListFragment = new ProductListFragment();
        historyFragment = new HistoryFragment();
        adapter.addFragment(productListFragment);
        adapter.addFragment(historyFragment);
        viewPager.setAdapter(adapter);
    }

    @Override
    public boolean onNavigationItemSelected(@NonNull MenuItem item) {

        switch (item.getItemId()){
            case R.id.navigation_home:
                viewPager.setCurrentItem(0);
                return true;
            case R.id.navigation_history:
                viewPager.setCurrentItem(1);
                return true;
        }
        return false;
    }



}

File: app/src/main/java/com/example/producttest/activity/HomeActivity.kt
package com.example.producttest.activity

import android.os.Bundle
import android.util.Log
import android.view.MenuItem
import android.view.View
import androidx.appcompat.app.AppCompatActivity
import androidx.fragment.app.FragmentStatePagerAdapter
import androidx.fragment.app.FragmentTransaction
import androidx.viewpager.widget.ViewPager
import androidx.viewpager.widget.ViewPager.OnPageChangeListener
import com.example.producttest.R
import com.example.producttest.adapter.ViewPagerAdapter
import com.example.producttest.databinding.ActivityHomeBinding
import com.example.producttest.fragments.CartFragment
import com.example.producttest.fragments.HistoryFragment
import com.example.producttest.fragments.ProductListFragment
import com.google.android.material.bottomnavigation.BottomNavigationView
import kotlinx.android.synthetic.main.activity_home.*

class HomeActivity : AppCompatActivity(), BottomNavigationView.OnNavigationItemSelectedListener {
    private lateinit var binding : ActivityHomeBinding
    private lateinit var viewPager: ViewPager

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = ActivityHomeBinding.inflate(layoutInflater)
        setContentView(binding.root)
        viewPager = findViewById(R.id.viewpager)
        binding.navViews.setOnNavigationItemSelectedListener(this)
        setupViewPager(viewPager)

    }

    fun getViewPager(): ViewPager? {
        if (null == viewPager) {
            viewPager = findViewById<View>(R.id.viewpager) as ViewPager
        }
        return viewPager
    }

    private fun setupViewPager(viewPager: ViewPager?) {
        val viewPagerAdapter = ViewPagerAdapter(supportFragmentManager, FragmentStatePagerAdapter.BEHAVIOR_RESUME_ONLY_CURRENT_FRAGMENT)
        viewPager!!.adapter = viewPagerAdapter
        viewPager!!.offscreenPageLimit = 1
        viewPager!!.addOnPageChangeListener(object : OnPageChangeListener {
            override fun onPageScrolled(position: Int, positionOffset: Float, positionOffsetPixels: Int) {}
            override fun onPageSelected(position: Int) {
                when (position) {
                    0 -> binding.navViews.menu.findItem(R.id.navigation_home).isChecked = true
                    1 -> binding.navViews.menu.findItem(R.id.navigation_history).isChecked = true
                }
            }

            override fun onPageScrollStateChanged(state: Int) {}
        })

    }

    override fun onNavigationItemSelected(item: MenuItem): Boolean {
        when (item.itemId) {
            R.id.navigation_home -> {
                viewPager!!.currentItem = 0

                return true
            }
            R.id.navigation_history -> {
                viewPager!!.currentItem = 1
//                val storyFragment: HistoryFragment = viewPager!!.adapter!!.instantiateItem(viewPager!!, 1) as HistoryFragment
//                storyFragment.ToastNe()
                return true
            }
        }
        return false
    }
}